similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -204,4 +214,36 @@ private long count(SolrQuery query)
         }
         return 0;
     }
+
+    private Collection<String> getPresentPhenotypeTerms(Patient patient)
+    {
+        Collection<String> termIds = new ArrayList<String>();
```

Comment:
@sdumitriu this was an intentional decision to include all ancestors multiple times to more closely mirror both the final scoring metric and the way patient phenotypes are indexed. A formal evaluation of which produces better results has not been done, but weighting ancestors more highly is actually good.